src/main/java/com/bobocode/hoverla/bring/context/BeanConfigClassValidator.java

```diff
@@ -0,0 +1,122 @@
+package com.bobocode.hoverla.bring.context;
+
+import com.bobocode.hoverla.bring.annotation.Bean;
+import com.bobocode.hoverla.bring.annotation.Configuration;
+import com.bobocode.hoverla.bring.exception.BeanConfigValidationException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Validates classes annotated with {@link Configuration @Configuration} and
+ * their methods marked with {@link Bean @Bean} that were scanned by {@link BeanScanner}.
+ *
+ * <p>Validation ensures:
+ * <ol>
+ *   <li>Valid type of a {@link Class} config object - it should be plain public class.
+ *   <br>See {@link BeanConfigClassValidator#checkClassModifiers(Class, List)} and {@link BeanConfigClassValidator#checkConfigIsPlainClass(Class, List)}</li>
+ *   <li>Valid bean methods - they should be public and non-static.
+ *   <br>See {@link BeanConfigClassValidator#checkMethodModifiers(Method, List)}</li>
+ *   <li>Valid constructor - bean config class should have a public non-args constructor.
+ *   <br>See {@link BeanConfigClassValidator#checkNoArgsConstructorExists(Class, List)}
+ * </ol>
+ *
+ * @see Bean
+ * @see Configuration
+ * @see BeanScanner
+ */
+@Slf4j
+class BeanConfigClassValidator {
+    private static final String VIOLATION_MSG_DELIMITER = "\n-";
+
+    /**
+     * Receives config class and checks violations of bean config class and its bean methods
+     *
+     * @param configClass target bean configuration class
+     */
+    void validate(Class<?> configClass) {
```

Comment:
Why is this method package-private?

Reply:
For the reason that it's not supposed to be public and ideally should be encapsulated in the package it's used. I'll return public since I believe we'll back to it in module story

Reply:
Fixed